File: ios/OutdoorApp/Sources/Models/Partner.swift
import Foundation

struct Partner: Codable {
    let id: Int?
    var name: String
    var avatarColor: String?
    var contact: String?
    var remark: String?
    var userId: Int?
    var createdAt: String?
    var updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, name
        case avatarColor = "avatar_color"
        case contact, remark
        case userId = "user_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct PartnerCreateReq: Codable {
    let name: String
    var avatarColor: String?
    var contact: String?
    var remark: String?

    enum CodingKeys: String, CodingKey {
        case name
        case avatarColor = "avatar_color"
        case contact, remark
    }

    init(name: String, avatarColor: String? = nil, contact: String? = nil, remark: String? = nil) {
        self.name = name
        self.avatarColor = avatarColor
        self.contact = contact
        self.remark = remark
    }
}

File: src/main/java/com/ruiyu/outdoor/model/Partner.java
package com.ruiyu.outdoor.model;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 伙伴实体
 */
@Data
@TableName("partners")
@AllArgsConstructor
@NoArgsConstructor
public class Partner {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 伙伴名称 */
    private String name;

    /** 伙伴头像颜色 */
    private String avatarColor;

    /** 联系方式 */
    private String contact;

    /** 备注 */
    private String remark;

    /** 所属用户ID */
    private Integer userId;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
